src/js/components/AppConfigJSONEditorComponent.jsx

```diff
@@ -0,0 +1,70 @@
+import AceEditor from "react-ace";
+import "brace/mode/json";
+import "brace/theme/monokai";
+import React from "react/addons";
+
+import AppDefaultFields from "../constants/AppDefaultFields";
+
+var AppConfigJSONEditorComponent = React.createClass({
+  displayName: "AppConfigJSONEditorComponent",
+
+  propTypes: {
+    app: React.PropTypes.object,
+    onChange: React.PropTypes.func.isRequired,
+    onError: React.PropTypes.func.isRequired
+  },
+
+  getDefaultProps: function () {
+    return {
+      app: null
+    };
+  },
+
+  shouldComponentUpdate: function () {
+    // avoid re-rendering the component as it causes the cursor to jump.
+    return false;
+  },
+
+  handleUpdate: function (value) {
+    let app = null;
+    let jsonWasValid = false;
+
+    try {
+      app = JSON.parse(value);
+      jsonWasValid = true;
+    }
+    catch (e) {
+      this.props.onError("Invalid JSON");
+    }
+
+    if (jsonWasValid) {
+      this.props.onChange(app);
+    }
+  },
+
+  getPrettyPrintedDefinition: function () {
+    // without this check, we would just show 'null' when creating a new app
+    let app = this.props.app == null
```

Comment:
Huh, guess I missed that part of the discussion. I'll make sure I update myself.